src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -40,25 +40,94 @@ class ThetaStarPathPlanner : public PathPlanner
                                  const std::vector<ObstaclePtr> &obstacles) override;
 
    private:
-    class Coordinate : public std::pair<unsigned int, unsigned int>
+    class Coordinate
     {
        public:
         Coordinate(unsigned int row, unsigned int col)
-            : std::pair<unsigned int, unsigned int>(row, col)
+            : row_(row), col_(col), internal_comparison_key_(computeComparisonKey(*this))
         {
         }
 
-        Coordinate() : std::pair<unsigned int, unsigned int>() {}
+        Coordinate() : row_(0), col_(0) {}
 
         unsigned int row(void) const
         {
-            return this->first;
+            return row_;
         }
 
         unsigned int col(void) const
         {
-            return this->second;
+            return col_;
         }
+
+        bool operator<(const Coordinate &other) const
+        {
+            return internal_comparison_key_ < other.internal_comparison_key_;
+        }
+
+        bool operator==(const Coordinate &other) const
+        {
+            return internal_comparison_key_ == other.internal_comparison_key_;
+        }
+
+        static unsigned long computeComparisonKey(const Coordinate &coord)
+        {
```

Comment:
missing javadocs and can we make this function private?

Reply:
Javadoc added. I made this function private but I add another getter for `CoordinatePair`